### ISSUE TITLE
Pipelines using pandas transformer output do not work with feature creation

### ISSUE THREAD

**Issue by admivsn** [feature-engine/feature_engine#670]
**Describe the bug**
Pipelines break when you use pandas output with feature creation, possibly due to feature_names_out.

**To Reproduce**
```
import pandas as pd

from feature_engine.creation import RelativeFeatures
from sklearn.pipeline import make_pipeline
from sklearn.dummy import DummyClassifier
from sklearn import set_config

set_config(transform_output="pandas")

X = pd.DataFrame({"feature_1": [1, 2, 3, 4, 5], "feature_2": [6, 7, 8, 9, 10]})
y = pd.Series([0, 1, 0, 1, 0])

pipeline = make_pipeline(
    RelativeFeatures(
        variables=["feature_1"],
        reference=["feature_2"],
        func=["div"]
    ),
    DummyClassifier()
)

pipeline.fit(X, y)
pipeline.predict(X)
```

**Comment by solegalli** [comment]
How annoying!

Thank you for reporting @admivsn 

I need to look into the inner workings of the pipeline.

By the looks of the error, the pipe is surprised to get the extra feature created by the RelativeFeatures.

 I'll see if I can dig some time over the weekend.

Cheers

**Comment by ClaudioSalvatoreArcidiacono** [comment]
The root cause of this issue is that at this line [here](https://github.com/feature-engine/feature_engine/blob/dade0f0df0158e5d53cad0db7aa5c4007e299415/feature_engine/creation/relative_features.py#LL197C1-L198C1):

```python
class RelativeFeatures(BaseCreation):
    ...
    def transform(self, X: pd.DataFrame) -> pd.DataFrame:
        X = super().transform(X)
```

The transform method of `BaseCreation` is called. This method will then call the `get_features_out` method defined in `GetFeatureNamesOutMixin` which will return the feature names out of the child class `RelativeFeatures` instead of the parent class `BaseCreation`  and since BaseCreation.transform(self, X) returns a DataFrame with 2 columns and `get_features_out` returns 3 columns this mismatch causes the error described in the issue.

A possible solution could be to make `BaseCreation.get_feature_names_out` return the input features, for example by implementing this function directly in `BaseCreation`

```python
from sklearn.utils.validation import _check_feature_names_in
class BaseCreation(BaseEstimator, TransformerMixin, GetFeatureNamesOutMixin):
    def get_feature_names_out(self, input_features):
        return _check_feature_names_in(self, input_features)
```

Not the most elegant of the solutions probably.

**Comment by solegalli** [comment]
Yes, I just figured that out. I am fixing as we speak. PR coming in the next minutes. Thank you!

That was a nasty one!
